### PR TITLE
Fix tailwindcss setup

### LINKED ISSUE #1301: `yarn rw setup tailwind` fails if packages already installed or if imports exist in `index.css`
Continued from https://github.com/redwoodjs/redwood/issues/1223

Note: `--force` option does not resolve the issue

Error output: 
```
$ yarn rw setup tailwind
yarn run v1.22.4
$ C:\Users\tobbe\dev\redwood\rw20\node_modules\.bin\rw setup tailwind
  √ Installing packages...
  √ Configuring PostCSS...
  × Initializing Tailwind CSS...
    → info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
    Adding imports to index.css...
    One more thing...
Command failed with exit code 1: yarn tailwindcss init
error Command "tailwindcss" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Done in 25.85s.
```

What's happening is that either:
- the installed packages are already present
...and/or...
- the tailwind imports already exist in `index.css`

Note: if either of the config files exist, the error output makes sense and states the issue accurately. But the error given above is confusing and inaccurate.

Way to reproduce:

1. Successfully run yarn rw setup tailwind
2. Remove the postcss config file rm web/config/postcss.config.js
3. Remove the tailwind config file rm web/tailwind.config.js
4. Run yarn rw setup tailwind again

From @Tobbe 
> Now it might seem like a strange/contrived thing to do to remove the config files. But say you're new, and screw something up with the config. You decide to delete the config files to try to start over again. It won't work.
>
> So I think yarn rw setup tailwind should re-create any missing config files (it does for the postcss config file already).
And it could also have a --force option to recreate/override config files even if they already exist

## How to improve
- In case of only packages already existing but no files or imports present, this command should work
- In case of imports already present, the command should fail but give accurate explanation
- Using `--force` should allow the command to run regardless of what is present

### PR DESCRIPTION
**Pull request by mohinderps**
Closes #1301 

This is draft PR, which fixed tailwindcss setup command. Feel free to review this and suggest the changes that need to done.

### CODE REVIEW

**Review by thedavidprice — APPROVED** [review]
🚀

**Comment by thedavidprice** [comment]
@mohinderps Woah, this is a ton of work and the functionality is 💯 Thank you for taking the time to get this one right. It's so helpful.

There's an incoming PR #1408 that will add an option to this command. The plan is to merge this first. However, I do want to loop in the author of #1408 to see what you've done and give any feedback before I merge (he's a great guy and has been helpful with Redwood since March). Once we have a coordinated plan, I'll merge. I expect this should be a quick turnaround.

Just let me know if you have any questions.

**Comment by thedavidprice** [comment]
Looping in @olance who's working on extending the command with an option via #1408 

I suggested a plan for us to discuss: 
- @olance helps review here to give input on code and coordinating with his PR
- we'll get this merged first
- then loop back to #1408 

Open to more discussion as needed. And, most of all, thank you both!

**Comment by mohinderps** [comment]
>@mohinderps Woah, this is a ton of work and the functionality is 💯 Thank you for taking the time to get this one right. It's so helpful.

Thank you @thedavidprice for such kind words. You have been of great help to me. I just followed your directions.
Looking forward to contribute more in this project.

And also if everything looks fine to you, can you also add hacktoberfest-accepted label ? So that this is counted as my hacktoberfest contribution.

**Comment by thedavidprice** [comment]
**re: `hacktoberfest-accepted`** 
If we merge the PR this week, it'll count without the label! And I’ll make sure we add the label otherwise. Thanks for asking.

**Comment by mohinderps** [comment]
Cool @thedavidprice !

**Comment by olance** [comment]
👋

Thanks for the ping @thedavidprice :)
I'll try to review this today!

File: packages/cli/src/commands/setup/tailwind/tailwind.js [hunk 1]
```diff
@@ -37,23 +37,58 @@ const tailwindImportsAndNotes = [
 
 const INDEX_CSS_PATH = path.join(getPaths().web.src, 'index.css')
 
+const tailwindImportsDoesExist = (indexCSS) => {
```
Comment (olance) [inline packages/cli/src/commands/setup/tailwind/tailwind.js:40]: Would just simplify this function name (which fixes a grammatical error at the same time ^^)

```suggestion
const tailwindImportsExist = (indexCSS) => {
```

File: packages/cli/src/commands/setup/tailwind/tailwind.js [hunk 2]
```diff
@@ -109,13 +163,20 @@ export const handler = async ({ force }) => {
...
         /**
          * Add tailwind imports and notes to the top of index.css
          */
         let indexCSS = fs.readFileSync(INDEX_CSS_PATH)
-        indexCSS = tailwindImportsAndNotes.join('\n') + indexCSS
-        fs.writeFileSync(INDEX_CSS_PATH, indexCSS)
+
+        const tailwindImportsExist = tailwindImportsDoesExist(indexCSS)
+
+        if (tailwindImportsExist) {
```
Comment (olance) [inline packages/cli/src/commands/setup/tailwind/tailwind.js:174]:
```suggestion
        if (tailwindImportsExist(indexCSS)) {
```

**Review by olance — APPROVED** [review]
Looks good to me! 👍 
Just added a suggestion on renaming `tailwindImportsDoesExist` to `tailwindImportsExist`, which does simplify the code a little bit as well as fixing a grammatical error ^^

**Comment by thedavidprice** [comment]
Merging this now. Thanks again @mohinderps 🚀

@olance ready for you to take over in #1408 🔥

**Pull request merged**